lib/jxl/exif.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
```

Comment:
This is encoder only, right? Could we name this enc_exif?

ResetExifOrientation() is only used in extras, are we gonna modify the exif if you request the pixels in some other orientation in the API? Looks like maybe this file should go to extras/ instead.

Reply:
Encoder-only atm, but maybe it could be useful to do ResetExifOrientation by default when we return Exif blobs in the decode api. If we just pass 'uncensored' Exif blobs, it's almost certain that many applications will interpret the Exif and incorrectly apply orientation twice.

Reply:
My request is that you don't add new calls from the tool/extras to internal API because I'm actually trying to do the opposite and it means just more tech debt that I have to clean up. Either we move this functionality to cjxl/djxl or the extras lib; or we implement it in the C API (which is in progress anyway).

Reply:
I think these functions should be integrated in the C api (called from the functions that get/set Exif metadata, conditioned on an option). Once cjxl/djxl use that api, they will no longer need to call these functions directly.

Reply:
For now, we need to fix a broken situation though: djxl was not writing correct exif metadata when outputting lossy jpeg, and problematic repeated orientation was happening when using cjxl/djxl to convert oriented jpg -> jxl -> png -> jxl.

Handling exif metadata properly in the api is something that also needs to be done, but atm it's not handled at all there so that's something else. Once it is handled there, we can integrate the exif handling stuff in the api and discuss if and what we want to expose there, but let's fix cjxl/djxl now already.